src/fileio.jl

```diff
@@ -60,11 +71,45 @@ function save(s::Stream{format"LAS"}, header::LasHeader, pointdata::Vector{T}) w
     # write header
     write(s, magic(format"LAS"))
     write(s, header)
-    bytes_togo = header.data_offset - position(s)
-    @assert bytes_togo == 0
 
     # write points
-    for i = 1:n
-        write(s, pointdata[i])
+    for p in pointdata
+        write(s, p)
+    end
+end
+
+function save(f::File{format"LAZ"}, header::LasHeader, pointdata::Vector{T}) where T <: LasPoint
+    # pipes las to laszip to write laz
+    open(`laszip -olaz -stdin -o "$(filename(f))"`, "w") do s
+        savebuf(s, header, pointdata)
+    end
+end
+
+# Uses a buffered write to the stream.
+# For saving to LAS this does not increase speed,
+# but it speeds up a lot when the result is piped to laszip.
```

Comment:
Would `BufferedStreams` help you here: https://github.com/BioJulia/BufferedStreams.jl rather than having to roll it yourself?